packages/rancher-vsphere-cpi/charts/templates/daemonset.yaml

```diff
@@ -44,20 +45,30 @@ spec:
                 values:
                 - "true"
       {{- end }}
+      tolerations:
+        # Rancher specific change: These tolerations are intentionally different from upstream to avoid lessening the scope to only NoSchedule with a specific key
+        # - key: node.cloudprovider.kubernetes.io/uninitialized
+        #   value: "true"
+        #   effect: NoSchedule
+        # - key: node-role.kubernetes.io/master
+        #   effect: NoSchedule
+        #   operator: Exists
+        # - key: node.kubernetes.io/not-ready
+        #   effect: NoSchedule
+        #   operator: Exists
+        - effect: NoExecute
+          operator: Exists
+        - effect: NoSchedule
+          operator: Exists
```

Comment:
This is the same problem we will encounter in https://github.com/rancher/rancher/issues/34751 for the CPI chart, right?

Could we either a) change this as part of this PR or b) create an issue to track the CPI issue?

Reply:
Sounds like is the same thing. I'll verify if we want it fixed in 2.5 since the fix for CSI got reverted. If we do, I'll implement it for both CPI and CSI. If we don't, I'll create an issue for CPI to track it separately

Reply:
I changed the tolerations to match upstream and added the option to override.